.github/workflows/deploy.yml

```diff
@@ -42,6 +42,8 @@ jobs:
             exit 1
           fi
       - uses: actions/checkout@v1
+      - name: Validate branch name
+        run: ./branchVal.sh $STAGE_PREFIX$branch_name
```

Comment:
Consider changing the name of branchVal.sh to something that doesn't contain "val".  I figure you mean val as in "validate"... no problems, just "val" is sort of a protected term around here, and usually denotes an environment.
validateBranchName.sh
branchNameValidation.sh
checkBranchNameIsValid.sh

Something like that.

Reply:
Also... calling this script effectively replaces the "Check branch name is a legal serverless stage name" step in deploy.yml   Be sure to remove that; I don't see it removed here.